src/WebJobs.Script.WebHost/Security/KeyManagement/SecretManager.cs

```diff
@@ -385,6 +385,12 @@ public async Task<(string, AuthorizationLevel)> GetAuthorizationLevelOrNullAsync
         {
             if (keyValue != null)
             {
+                if (_authorizationCache.TryGetValue(keyValue, out (string, AuthorizationLevel) value))
```

Comment:
The authorization of a token involves a lot of overhead. We can do this once and cache the result, as long as we ensure that we clear this cache when secrets change.